File: desktop/src/main/java/org/gipsybuho/recetasfamiliares/core/SoundLevel.java
package org.gipsybuho.recetasfamiliares.core;

/** Cuanto sonido quiere el usuario. Se elige en Ajustes. */
public enum SoundLevel {

    /** Ninguno. Es el valor por defecto: la aplicacion nace callada. */
    SILENCIO,
    /** Solo lo que tiene consecuencias: guardado, error, borrado, avisos, cocina. */
    IMPORTANTES,
    /** Tambien la navegacion y los cambios de estado menores. */
    TODOS;

    public boolean allows(SoundEffect effect) {
        return switch (this) {
            case SILENCIO -> false;
            case IMPORTANTES -> effect.isImportant();
            case TODOS -> true;
        };
    }

    /** Tolerante a valores corruptos o de una version futura. */
    public static SoundLevel fromPreference(String stored, SoundLevel fallback) {
        if (stored == null || stored.isBlank()) {
            return fallback;
        }
        try {
            return valueOf(stored.trim().toUpperCase(java.util.Locale.ROOT));
        } catch (IllegalArgumentException desconocido) {
            return fallback;
        }
    }

    /**
     * La preferencia antigua era un booleano. Quien lo tuviera activado pasa a
     * IMPORTANTES: ni se queda mudo de golpe ni empieza a oirlo todo.
     */
    public static SoundLevel fromLegacyEnabled(boolean enabled) {
        return enabled ? IMPORTANTES : SILENCIO;
    }
}

File: android/app/src/main/java/org/gipsybuho/recetasfamiliares/core/SoundLevel.kt
package org.gipsybuho.recetasfamiliares.core

/** Cuanto sonido quiere el usuario. Se elige en Perfil. */
enum class SoundLevel {

    /** Ninguno. Valor por defecto: la aplicacion nace callada. */
    SILENCIO,
    /** Solo lo que tiene consecuencias: guardado, error, borrado, avisos, cocina. */
    IMPORTANTES,
    /** Tambien la navegacion y los cambios de estado menores. */
    TODOS;

    fun allows(effect: SoundEffect): Boolean = when (this) {
        SILENCIO -> false
        IMPORTANTES -> effect.important
        TODOS -> true
    }

    companion object {
        /** Tolerante a valores corruptos o de una version futura. */
        fun fromPreference(stored: String?, fallback: SoundLevel): SoundLevel {
            if (stored.isNullOrBlank()) return fallback
            return runCatching { valueOf(stored.trim().uppercase()) }.getOrDefault(fallback)
        }
    }
}
